core/arch/arm/plat-hikey/main.c

```diff
@@ -73,6 +73,9 @@ register_phys_mem(MEM_AREA_IO_NSEC, PMX1_BASE, PMX1_REG_SIZE);
 register_phys_mem(MEM_AREA_IO_NSEC, GPIO6_BASE, PL061_REG_SIZE);
 register_phys_mem(MEM_AREA_IO_NSEC, SPI_BASE, PL022_REG_SIZE);
 #endif
+#if defined(CFG_DYN_SHM)
+register_nsec_ddr(DRAM0_BASE, DRAM0_SIZE_NSEC);
```

Comment:
Does it matter if this is always registered?

Reply:
This is to be able to test the previous SHM. If I understand correctly, as soon as some NSEC memory is registered here, the older SHM pool mechanism is not used anymore?

Reply:
Both can actually be used in parallel.

Reply:
Mmm. I thought that `TEEC_AllocateSharedMemory()` would not get its buffers from the static SHM pool anymore when the dynamic capability is advertised by OP-TEE? https://github.com/OP-TEE/optee_client/blob/master/libteec/src/tee_client_api.c#L726:L751

Reply:
I believe @jenswi-linaro wanted to say, that if you are using old kernel driver, old older client library, or both at the same time, they will fallback to static SHM.

Reply:
Ah, yes, that I understand, and it's good.
My problem is: given one version of the code (because it is too complex to test compatibility with older code... we don't do it currently), I would like to make sure the "old" shm code still works. This CFG_ is an easy way.

And, some users might want to use only the older scheme because it doesn't suffer from the boomerang (aka semantic gap) weakness -- or to a lesser extent at least.

Reply:
To disable the new scheme we should disable it from entry functions for two reasons:
1. it's more clear what's going on
2. some code can be optimized away by the linker

Reply:
You mean, it should be taken care of in https://github.com/OP-TEE/optee_os/pull/1631? I think that would be better, too. And avoid inconsistencies between platforms.

Reply:
Yes